packages/material-ui-lab/src/internal/pickers/Picker/makePickerWithState.tsx

```diff
@@ -88,13 +93,10 @@ export function makePickerWithState<
         />
       </WrapperComponent>
     );
-  }
+  });
 
-  // tslint:disable-next-line
-  // @ts-ignore Simply ignore generic values in props, because it is impossible
-  // to keep generics without additional cast when using forwardRef
-  // @see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/35834
-  return React.forwardRef<HTMLInputElement, React.ComponentProps<typeof PickerWithState>>(
-    (props, ref) => <PickerWithState {...(props as any)} forwardedRef={ref} />,
-  );
+  // @ts-expect-error Types are equal except `forwardRef` calls the returned types with `PropsWithoutRef`.
+  // The distributive nature of `PropsWithOutRef` causes the type error.
+  // TODO: Find out why we need a distributive `PropsWithOutRef`.
+  return PickerWithState as PickerComponent<T, TWrapper>;
```

Comment:
Removing this extra (anonymous) render function was the main goal. The rest followed naturally.